### ISSUE TITLE
Pre-build plugin

### ISSUE THREAD

**Issue by AlexAntonides** [CONNECT-platform/codedoc#141]
I'm a bit confused on how to build plugins, because the [documentation](https://codedoc.cc/docs/config/plugins) only provides an example on how to add scripts to the website. 

All in all, I'd like to generate one documentation website from multiple repositories. To do this, my plan was to generate markdown documentation files for each repository using TSDoc/PyDoc depending on the repository, and then let CodeDoc compile these markdown files into one giant documentation site. Unfortunately, this won't work well, because the generated markdown files contain anchor points pointing to other .md files, which the Codedoc parser doesn't detect. In other words, I'll end up with a website with parsed .md files, but all anchor links still point to the .md files, instead of the compiled .html files. 
To fix this, I'd like to create a plugin that allows me to change the parsing in such a way, that each anchor link that is a relative path pointing to a .md file, gets changed into a .html extension.  

In other words, I'd like to be able to write a plugin that 'hooks' into the building process and add functionality to the parser.

**Comment by loreanvictor** [comment]
Hi! Yes perhaps more examples could help with this.

What you can do is to process each HTML file and modify its links (for example, if they are relative and end with `.md`, then change them accordingly). For that purpose, you can use [post processor functions](https://codedoc.cc/docs/config/page#post-processing). Here is an example of a [simple post processor that adds line numbers to code snippets](https://github.com/CONNECT-platform/codedoc/blob/master/src/components/code/line-links/add-line-ids.ts), used by CODEDOC internally. You can also create (and publish) an independent plugin by following [the example provided in the docs](https://codedoc.cc/docs/config/plugins#writing-a-plugin), or by taking a look at [this actual external plugin](https://github.com/CONNECT-platform/coding-blog-plugin).

Note that, as [mentioned here in the docs](https://codedoc.cc/docs/config/page#post-processing), your post processor function will be provided an `HTMLDocument` object, so you can mutate its elements in-place:

```ts
export function myPostProcessor(html: HTMLDocument) {
  html.querySelectorAll('a').forEach(link => {
    if (link.href matches some criteria) {
      link.setAttribute('href', the fixed URL)
    }
  })
}
```
